addons/rmsmartshape/shapes/shape.gd

```diff
@@ -1736,13 +1762,70 @@ func _build_edge_with_material(
 		for q in new_quads:
 			edge.quads.push_back(q)
 		i += next_point_delta
-
+	
+	# leftover final taper for the last sharp corner if required
+	if taper_sharp:
+		if sharp_taper_next != null && edge.quads[0].corner == SS2D_Quad.CORNER.NONE:
+			var taper := _taper_quad_right(sharp_taper_next, edge_material, texture_idx)
+			if taper != null:
+				taper.ignore_weld_next = true
+				edge.quads.push_back(taper)
+			else:
+				sharp_taper_next.ignore_weld_next = true
+		sharp_taper_next = null
+	
 	if edge_material_meta != null:
 		if edge_material_meta.weld:
 			_weld_quad_array(edge.quads, edge.wrap_around)
 
 	return edge
 
+func _taper_quad_left(new_quad: SS2D_Quad, edge_material: SS2D_Material_Edge, texture_idx: int) -> SS2D_Quad:
```

Comment:
`_taper_quad_left` and `_taper_quad_right` are largely identical. Please try to share common code or merge both functions into one with an additional argument indicating whether it should be left or right.